ckanext/data_qld/resources_plugin.py

```diff
@@ -90,6 +98,10 @@ def before_create(self, context, data_dict):
     def before_update(self, context, current_resource, updated_resource):
         return self.check_file_upload(updated_resource)
 
+    def before_show(self, resource_dict):
+        if request.path_info.split('/')[1] == 'api':
```

Comment:
FYI You can also use toolkit.get_endpoint() which returns a tuple (controller|blueprint, action|view).
As far as I am aware any api calls should return `toolkit.get_endpoint()[1] == 'action'`

Reply:
refactored